src/oss-find-squats-lib/MutateExtension.cs

```diff
@@ -0,0 +1,109 @@
+            new DuplicatorMutator(),
+            new PrefixMutator(),
+            new RemovedCharacterMutator(),
+            new SeparatorMutator(),
+            new SuffixMutator(),
+            new SwapOrderOfLettersMutator(),
+            new UnicodeHomoglyphMutator(),
+            new VowelSwapMutator()
+        };
+
+        internal static IEnumerable<Mutator> NugetMutators { get; } = BaseMutators.Where(x => x is not UnicodeHomoglyphMutator and not SuffixMutator)
+            .Append(new SuffixMutator(additionalSuffixes: new[] { "net", ".net", "nuget" }, skipSuffixes: new[] { "." }));
+
+        internal static IEnumerable<Mutator> NpmMutators { get; } = BaseMutators.Where(x => x is not UnicodeHomoglyphMutator and not SuffixMutator)
+            .Concat(new Mutator[]
+                {
+                    new SubstitutionMutator(new List<(string Original, string Substitution)>()
+                    {
+                        ("js", "javascript"),
+                        ("ts", "typescript"),
+                    }),
+                    new SuffixMutator(additionalSuffixes: new[] { "js", ".js", "javascript", "ts", ".ts", "typescript"})
+                });
+
+        public static IEnumerable<Mutator> GetDefaultMutators(this BaseProjectManager manager) => manager switch
+        {
+            NuGetProjectManager => NugetMutators,
+            NPMProjectManager => NpmMutators,
+            _ => BaseMutators
+        };
+
+        public static async IAsyncEnumerable<FindSquatResult> EnumerateSquats(this BaseProjectManager manager, PackageURL purl, MutateOptions options)
+        {
+            await foreach(var mutation in manager.EnumerateSquats(purl, manager.GetDefaultMutators(), options))
+            {
+                yield return mutation;
+            }
+        }
+
+        public static async IAsyncEnumerable<FindSquatResult> EnumerateSquats(this BaseProjectManager manager, PackageURL purl, IEnumerable<Mutator> mutators, MutateOptions options)
+        {
+            HashSet<string> alreadyChecked = new();
+            if (purl.Name is null || purl.Type is null)
+            {
+                yield break;
+            }
+            foreach(var mutator in mutators)
+            {
+                foreach(var mutation in mutator.Generate(purl.Name))
+                {
+                    if (!alreadyChecked.Add(mutation.Mutated))
+                    {
+                        Logger.Trace($"Already chcked {mutation.Mutated}. Skipping.");
+                        continue;
+                    }
+                    if (options.SleepDelay > 0)
+                    {
+                        Thread.Sleep(options.SleepDelay);
+                    }
+                    var candidatePurl = new PackageURL(purl.Type, mutation.Mutated);
+                    FindSquatResult? res = null;
+                    try
+                    {
+                        var versions = await manager.EnumerateVersions(candidatePurl);
```

Comment:
I don't know that we can rely on all repositories getting a 404 when a package doesn't exist. For example if we are querying and the result would be json it's possible the response would just an empty valid json object right?

Reply:
I took a quick look at the methods we have implemented so far and I think enumerate versions is still going to be the most robust one.  However, I have added a "PackageExists" method to the BaseProjectManager which (for now) just wraps the EnumerateVersions(purl).Any() check.  The squat checked will use the PackageExists method, which can be optimized in the future.

Reply:
If you can validate that your solution of just a getasync works for all (or many) managers it can override the call for PackageExists for those managers.

Reply:
Check the latest. I have made a custom implementation for Nuget  and NPM managers that will shortcut enumerating versions. By default a package manager will enumerate versions unless it has a custom implementation for PackageExists.

Reply:
#283 to track implementation for the remaining managers.